### PR TITLE
Adapt default+schedule for YaST cases for x86_64 online part1

### PR DESCRIPTION
**Pull request by hjluo**
Adapt default+schedule for YaST online cases for x86_64 part 1.

- Related ticket:  https://progress.opensuse.org/issues/119887
- Related MR:  https://gitlab.suse.de/qsf-y/qa-sle-functional-y/-/merge_requests/450
- Needles: N/A
- Verification run:  [YaST_Online_Part1](https://openqa.suse.de/tests/overview?distri=sle&version=15-SP5&build=hjluo%2Fos-autoinst-distri-opensuse%23x86_def_online_p1)

### CODE REVIEW

File: schedule/yast/btrfs/btrfs+warnings_x86_64.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+name: btrfs+warnings
```
Comment (coolgw) [inline schedule/yast/btrfs/btrfs+warnings_x86_64.yaml:2]: Why we need create new yaml file with suffix _x86_64?  Should we share same yaml file together with default file contain different suffix _86/s390?

**Comment by tinawang123** [comment]
If we need follow different default yaml files for each arch and one case only have one schedule yaml file policy. We need change the schedule yaml file's name. Like: schedule/yast/btrfs/btrfs+warnings_x86_64.yaml  --> schedule/yast/btrfs/btrfs+warnings.yaml  WDYT?

**Comment by hjluo** [comment]
> If we need follow different default yaml files for each arch and one case only have one schedule yaml file policy. We need change the schedule yaml file's name. Like: schedule/yast/btrfs/btrfs+warnings_x86_64.yaml --> schedule/yast/btrfs/btrfs+warnings.yaml WDYT?

The thing is that for example schedule/yast/btrfs/btrfs+warnings.yaml was used both on x86_64 and Aarch, the current fix is just to add x86_64 suffix for x86_64 while don't break Aarch64.  Maybe we have ticket to unify them as one once we have Aarch64 default?
@jknphy  Can you explain that to Tina and Wei? Thanks a lot.

**Comment by jknphy** [comment]
> > If we need follow different default yaml files for each arch and one case only have one schedule yaml file policy. We need change the schedule yaml file's name. Like: schedule/yast/btrfs/btrfs+warnings_x86_64.yaml --> schedule/yast/btrfs/btrfs+warnings.yaml WDYT?
> 
> The thing is that for example schedule/yast/btrfs/btrfs+warnings.yaml was used both on x86_64 and Aarch, the current fix is just to add x86_64 suffix for x86_64 while don't break Aarch64. Maybe we have ticket to unify them as one once we have Aarch64 default? @jknphy Can you explain that to Tina and Wei? Thanks a lot.

yes, the idea is work independently for each arch and not break anything, and later will be easy to unify test suite per test suite refactoring the parts that we need.

**Comment by jknphy** [comment]
Hi @hjluo to speed-up this task @rakoenig provided some necessary fixes in #16218 , please update schedules accordingly.

File: schedule/yast/encryption/cryptlvm_iscsi.yaml [hunk 2]
```diff
@@ -11,40 +11,34 @@ vars:
...
-  - installation/boot_encrypt
-  - installation/first_boot
-  - console/system_prepare
-  - console/hostname
-  - console/force_scheduled_tasks
-  - installation/validation/ibft
-  - console/validate_lvm
-  - console/validate_encrypt
+  registration:
+    - installation/iscsi_configuration
```
Comment (jknphy) [inline schedule/yast/encryption/cryptlvm_iscsi.yaml:15]: this is not related with registration, should overwrite `disk_activation` in default file.

File: schedule/yast/gpt.yaml [hunk 3]
```diff
@@ -5,32 +5,19 @@ description: >
...
-  - console/validate_partition_table_via_blkid
-  - console/validate_blockdevices
+  extension_module_selection:
+    - installation/module_registration/register_module_desktop
+  add_on_product:
+    - installation/add_on_product/skip_install_addons
+  system_role:
+    - installation/system_role/accept_selected_role_SLES_with_GNOME
+  default_systemd_target:
+    - installation/change_desktop
```
Comment (jknphy) [inline schedule/yast/gpt.yaml:15]: change_desktop should be under `software`.

File: schedule/yast/ssh_keys/do_not_import_ssh_keys.yaml [hunk 4]
```diff
@@ -7,30 +7,21 @@ vars:
...
+  extension_module_selection:
+    - installation/module_registration/register_module_desktop
+  add_on_product:
+    - installation/add_on_product/skip_install_addons
+  system_role:
+    - installation/system_role/accept_selected_role_SLES_with_GNOME
+  default_systemd_target:
+    - installation/installation_settings/validate_default_target
+  installation_settings:
+    - installation/ssh_host_key_settings/do_not_import_ssh_host_keys
```
Comment (rakoenig) [inline schedule/yast/ssh_keys/do_not_import_ssh_keys.yaml:23]: I wonder if we could put `do_not_import_ssh_host_keys` into the `security` section. WDYT?

File: schedule/yast/repo_inst.yaml [hunk 5]
```diff
@@ -10,28 +10,17 @@ vars:
...
-  - installation/first_boot
-  - console/validate_mirror_repos
+  product_selection:
+    - installation/validation/repo_inst
+    - installation/validation/validate_install_repo
+    - installation/product_selection/install_SLES
+  extension_module_selection:
+    - installation/module_registration/register_module_desktop
+  add_on_product:
+    - installation/add_on_product/skip_install_addons
```
Comment (rakoenig) [inline schedule/yast/repo_inst.yaml:20]: In the adaption in [PR#16218](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/16218) this will be the default already.

File: schedule/yast/lvm_raid1/lvm+raid1_sle15_x86_64.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+vars:
+  RAIDLEVEL: 1
+  LVM: 1
+  YUI_REST_API: 1
+schedule:
+  extension_module_selection:
+    - installation/module_registration/skip_module_registration
+  add_on_product:
+    - installation/add_on_product/skip_install_addons
+  suggested_partitioning:
```
Comment (rakoenig) [inline schedule/yast/lvm_raid1/lvm+raid1_sle15_x86_64.yaml:15]: In the adaption in [PR#16218](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/16218) this will be the default already.

File: schedule/yast/lvm_multipath_x86_64.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+vars:
+  DESKTOP: textmode
+  MULTIPATH: 1
+  LVM: 1
+  SEPARATE_HOME: 0
+  HDDMODEL: scsi-hd
+  YUI_REST_API: 1
+schedule:
+  extension_module_selection:
+    - installation/module_registration/skip_module_registration
```
Comment (rakoenig) [inline schedule/yast/lvm_multipath_x86_64.yaml:14]: In the adaption in [PR#16218](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/16218) this will be the default already.

File: schedule/yast/lvm_multipath_x86_64.yaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  add_on_product:
+    - installation/multipath
+    - installation/add_on_product/skip_install_addons
+  suggested_partitioning:
+    - installation/partitioning/select_guided_setup
+    - installation/partitioning/guided_setup/enable_lvm
+    - installation/partitioning/guided_setup/do_not_propose_separate_home
```
Comment (rakoenig) [inline schedule/yast/lvm_multipath_x86_64.yaml:21]: The 3 lines above should be tied to the `guided_partitioning` section that was added in PR#16218.

File: schedule/yast/lvm_multipath_encrypted_x86_64.yaml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+schedule:
+  extension_module_selection:
+    - installation/module_registration/skip_module_registration
+  add_on_product:
+    - installation/multipath
+    - installation/add_on_product/skip_install_addons
+  suggested_partitioning:
+    - installation/partitioning/select_guided_setup
+    - installation/partitioning/guided_setup/encrypt_lvm_simple_pwd
+    - installation/partitioning/guided_setup/accept_default_fs_options
```
Comment (rakoenig) [inline schedule/yast/lvm_multipath_encrypted_x86_64.yaml:22]: The 3 lines above should be tied to the `guided_partitioning` section added in PR#16218.

File: schedule/yast/lvm/lvm_thin_provisioning_x86_64.yaml [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+  2 LVM and BIOS boot partitions. Thin pool and thin lv reside on the second
+  LVM partition, where /home (XFS) is being mounted. Partitioning is validated
+  in the booted system after the installation.
+vars:
+  YUI_REST_API: 1
+schedule:
+  extension_module_selection:
+    - installation/module_registration/register_module_desktop
+  add_on_product:
+    - installation/add_on_product/skip_install_addons
```
Comment (rakoenig) [inline schedule/yast/lvm/lvm_thin_provisioning_x86_64.yaml:14]: In the adaption in [PR#16218](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/16218) this will be the default already.

File: schedule/yast/lvm/lvm_sle_x86_64.yaml [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+  DESKTOP: textmode
+  LVM: '1'
+  MAX_JOB_TIME: '14400'
+  OPT_KERNEL_PARAMS: console=%SERIALDEV% VNCSize=1024x768
+  YUI_REST_API: 1
+schedule:
+  extension_module_selection:
+    - installation/module_registration/skip_module_registration
+  add_on_product:
+    - installation/add_on_product/skip_install_addons
```
Comment (rakoenig) [inline schedule/yast/lvm/lvm_sle_x86_64.yaml:17]: In the adaption in [PR#16218](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/16218) this will be the default already.

File: schedule/yast/lvm/lvm+resize_lv.yaml [hunk 12]
```diff
@@ -9,38 +9,26 @@ vars:
...
-  - installation/first_boot
-  - console/system_prepare
-  - console/check_network
-  - console/prepare_test_data
-  - console/consoletest_setup
-  - locale/keymap_or_locale
-  - console/consoletest_finish
-  - console/validate_modify_existing_partition
+  add_on_product:
+    - installation/add_on_product/skip_install_addons
```
Comment (rakoenig) [inline schedule/yast/lvm/lvm+resize_lv.yaml:15]: In the adaption in [PR#16218](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/16218) this will be the default already.

File: schedule/yast/lvm/lvm+resize_lv.yaml [hunk 13]
```diff
@@ -9,38 +9,26 @@ vars:
...
+  suggested_partitioning:
+    - installation/partitioning/select_guided_setup
+    - installation/partitioning/guided_setup/enable_lvm
+    - installation/partitioning/guided_setup/do_not_propose_separate_home
+    - installation/partitioning/resize_existing_lv
```
Comment (rakoenig) [inline schedule/yast/lvm/lvm+resize_lv.yaml:20]: The 3 lines above should be tied to the `guided_partitioning` section added in PR#16218.

File: schedule/yast/iscsi_ibft.yaml [hunk 14]
```diff
@@ -9,33 +9,25 @@ vars:
...
-  - console/force_scheduled_tasks
-  - shutdown/grub_set_bootargs
-  - installation/validation/ibft
+  registration:
+    - installation/iscsi_configuration
+    - installation/registration/register_via_scc
+  extension_module_selection:
+    - installation/module_registration/register_module_desktop
+  add_on_product:
+    - installation/add_on_product/skip_install_addons
```
Comment (rakoenig) [inline schedule/yast/iscsi_ibft.yaml:18]: In the adaption in [PR#16218](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/16218) this will be the default already.

File: schedule/yast/encryption/cryptlvm_iscsi.yaml (hunk 2 above)
Comment (rakoenig) [reply schedule/yast/encryption/cryptlvm_iscsi.yaml:15]: I added `disk_activation` to the `default_x86_64.yaml` file in PR#16218.

**Comment by hjluo** [comment]
> Hi @hjluo to speed-up this task @rakoenig provided some necessary fixes in #16218 , please update schedules accordingly.

OK, thanks, I'd update this PR accordingly.

File: schedule/yast/encryption/cryptlvm_iscsi.yaml (hunk 2 above)
Comment (hjluo) [reply schedule/yast/encryption/cryptlvm_iscsi.yaml:15]: OK

File: schedule/yast/ssh_keys/do_not_import_ssh_keys.yaml (hunk 4 above)
Comment (hjluo) [reply schedule/yast/ssh_keys/do_not_import_ssh_keys.yaml:23]: changed.

File: schedule/yast/repo_inst.yaml (hunk 5 above)
Comment (hjluo) [reply schedule/yast/repo_inst.yaml:20]: OK

File: schedule/yast/lvm_raid1/lvm+raid1_sle15_x86_64.yaml (hunk 6 above)
Comment (jknphy) [reply schedule/yast/lvm_raid1/lvm+raid1_sle15_x86_64.yaml:11]: here we have 3 cases, expert, guided and suggested(like default) but we can think later if we need three keys or overwrite one key called 'partitioning', perhaps is too soon, so we are good I think.

File: schedule/yast/lvm/lvm_sle_x86_64.yaml [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+  wizard screen.
+name: cryptlvm
+vars:
+  DESKTOP: textmode
+  LVM: '1'
+  MAX_JOB_TIME: '14400'
+  OPT_KERNEL_PARAMS: console=%SERIALDEV% VNCSize=1024x768
+  YUI_REST_API: 1
+schedule:
+  suggested_partitioning:
```
Comment (jknphy) [inline schedule/yast/lvm/lvm_sle_x86_64.yaml:14]: `guided_partitioning` instead.

File: schedule/yast/lvm_multipath_x86_64.yaml [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+  MULTIPATH: 1
+  LVM: 1
+  SEPARATE_HOME: 0
+  HDDMODEL: scsi-hd
+  YUI_REST_API: 1
+schedule:
+  extension_module_selection:
+    - installation/module_registration/skip_module_registration
+  add_on_product:
+    - installation/multipath
```
Comment (jknphy) [inline schedule/yast/lvm_multipath_x86_64.yaml:16]: this deserve an additional key in default file, ie: multipath.

File: schedule/yast/lvm_multipath_x86_64.yaml [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+  guided_partitioning:
+    - installation/partitioning/select_guided_setup
+    - installation/partitioning/guided_setup/enable_lvm
+    - installation/partitioning/guided_setup/do_not_propose_separate_home
+  default_systemd_target:
+    - installation/installation_settings/validate_default_target
+  system_validation:
+    - console/validate_multipath
+    - console/validate_lvm
+    - console/hostname
```
Comment (jknphy) [inline schedule/yast/lvm_multipath_x86_64.yaml:27]: these  3 test modules, hostname...force_schedule_tasks, should be under system_preparation, it was wrong in the original code, could you please change it?

File: schedule/yast/lvm_multipath_encrypted_x86_64.yaml [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+  DESKTOP: textmode
+  MULTIPATH: 1
+  LVM: 1
+  ENCRYPT: 1
+  SEPARATE_HOME: 0
+  HDDMODEL: scsi-hd
+  YUI_REST_API: 1
+schedule:
+  add_on_product:
+    - installation/multipath
```
Comment (jknphy) [inline schedule/yast/lvm_multipath_encrypted_x86_64.yaml:15]: deserve extra key in default.

File: schedule/yast/lvm_multipath_encrypted_x86_64.yaml [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+    - installation/add_on_product/skip_install_addons
+  guided_partitioning:
+    - installation/partitioning/select_guided_setup
+    - installation/partitioning/guided_setup/encrypt_lvm_simple_pwd
+    - installation/partitioning/guided_setup/accept_default_fs_options
+    - installation/partitioning/accept_proposed_layout
```
Comment (jknphy) [inline schedule/yast/lvm_multipath_encrypted_x86_64.yaml:21]: this one would be repeated, you need to drop it.

File: schedule/yast/btrfs/btrfs+warnings_x86_64.yaml [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+  something is missing during manual partitioning using Expert Partitioner.
+  Following warning are verified:
+    - Missing root partition;
+    - Minimal size for the root with btrfs and snapshots
+    - Missing boot partition (bios boot,/boot/zipl/,EFI, prep-boot).
+vars:
+  FILESYSTEM: btrfs
+  YUI_REST_API: 1
+schedule:
+  suggested_partitioning:
```
Comment (jknphy) [inline schedule/yast/btrfs/btrfs+warnings_x86_64.yaml:14]: should be under `expert_partitioning`

File: schedule/yast/btrfs/btrfs+warnings_x86_64.yaml [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+    - installation/partitioning/warning/no_root
+    - installation/partitioning/warning/snapshots_small_root
+    - installation/partitioning/warning/no_boot
+    - installation/partitioning/warning/boot_small_for_kernel
+    - installation/partitioning/warning/bios_boot_small_for_bootloader
+    - installation/partitioning/warning/prep_small
+    - installation/partitioning/warning/zipl_small
+    - installation/partitioning/warning/rootfs_small
+    - installation/partitioning/accept_proposed_layout
```
Comment (jknphy) [inline schedule/yast/btrfs/btrfs+warnings_x86_64.yaml:23]: we need to drop this line once is under suggested_partitioning.

File: schedule/yast/iscsi_ibft.yaml [hunk 22]
```diff
@@ -9,33 +9,23 @@ vars:
...
-  - shutdown/grub_set_bootargs
-  - installation/validation/ibft
+  registration:
+    - installation/iscsi_configuration
+    - installation/registration/register_via_scc
+  extension_module_selection:
+    - installation/module_registration/register_module_desktop
+  system_role:
+    - installation/system_role/accept_selected_role_SLES_with_GNOME
+  suggested_partitioning:
```
Comment (jknphy) [inline schedule/yast/iscsi_ibft.yaml:19]: this should be under `guided_partitioning` dropping last test modules `accept_proposed_layout` to avoid repetition.

File: schedule/yast/lvm/lvm_thin_provisioning_x86_64.yaml [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+  LVM partition, where /home (XFS) is being mounted. Partitioning is validated
+  in the booted system after the installation.
+vars:
+  YUI_REST_API: 1
+schedule:
+  extension_module_selection:
+    - installation/module_registration/register_module_desktop
+  system_role:
+    - installation/system_role/accept_selected_role_SLES_with_GNOME
+  suggested_partitioning:
```
Comment (jknphy) [inline schedule/yast/lvm/lvm_thin_provisioning_x86_64.yaml:15]: expert_partitioning

File: schedule/yast/lvm/lvm+resize_lv.yaml (hunk 13 above)
Comment (jknphy) [reply schedule/yast/lvm/lvm+resize_lv.yaml:16]: looks like this test module include to accept the suggested partitioning, so we would be doing it twice, as it is out of scope change the code, I guess we need to clear up:
`suggested_partitioning: []`

File: schedule/yast/iscsi_ibft.yaml [hunk 24]
```diff
@@ -9,33 +9,21 @@ vars:
...
-  - shutdown/grub_set_bootargs
-  - installation/validation/ibft
+  disk_activation:
+    - installation/iscsi_configuration
+  extension_module_selection:
+    - installation/module_registration/register_module_desktop
+  system_role:
+    - installation/system_role/accept_selected_role_SLES_with_GNOME
+  guided_partitioning:
+    - installation/partitioning
```
Comment (jknphy) [inline schedule/yast/iscsi_ibft.yaml:19]: this one seems wrong in the original schedule, could be replaced by:
`- installation/partitioning/select_guided_setup`

File: schedule/yast/installer_extended/installer_extended_x86_64.yaml [hunk 25]
```diff
@@ -0,0 +1,88 @@
...
+    - installation/licensing/verify_license_has_to_be_accepted
+    - installation/licensing/accept_license
+  extension_module_selection:
+    - installation/module_registration/view_development_versions
+    - installation/module_registration/verify_module_registration
+    - installation/module_registration/skip_module_registration
+  clock_and_timezone:
+    - installation/releasenotes
+    - installation/clock_and_timezone/accept_timezone_configuration
+  default_systemd_target:
```
[the remark this answers is not in the document]
Comment (jknphy) [reply schedule/yast/installer_extended/installer_extended_x86_64.yaml:38]: please check this: https://openqa.suse.de/tests/10436396#step/root_simple_pwd/1
